### ISSUE TITLE
Questions about the codes

### ISSUE THREAD

**Issue by nightwork23** [NahuelCostaCortez/Remaining-Useful-Life-Estimation-Variational#6]
Hello, I'm really interested in your work and have tried to reproduce it for many days,but I can't do it.I am curious about three questions.

1.https://github.com/NahuelCostaCortez/Remaining-Useful-Life-Estimation-Variational/blob/main/RULRVE.ipynb
For your example above ,I run it both in CPU and GPU devices,the best result is:
test set RMSE:13.93,SCORE:326.35,
which is lower than the result in your paper. It is also lower than Semi-supervised[23],DCNN[55],MS-DCNN[55] listed in your paper.

2.When I tried to execute the code main.py for FD002 dataset,the result is :
test set RMSE:26.15,SCORE:14940,
which is a bad result and seems wrong. Is there a problem with my settings or other reasons?

3.Through printing the model of h5 files, I find that the last layer of the encoder is 'lambda', which is different from the  'sampling' layer in model.py.
I wonder if you have improved your work about the design of encoder.

**Comment by NahuelCostaCortez** [comment]
Hi @nightwork23,

Thanks for your interest in my work!

Note that the [example](https://github.com/NahuelCostaCortez/Remaining-Useful-Life-Estimation-Variational/blob/main/RULRVE.ipynb) trains the model with just 10 epochs. If you want to achieve the best possible result you may want to configure EarlyStopping and ModelCheckpoint callbacks and let it train until the val loss no longer improves.

As for the results for FD002, if you are using the sensors I left in the example you are lacking one sensor. Try with ['s_3', 's_4', 's_7', 's_10', 's_11', 's_12'].

The 'lambda' layer it is because I first included the sampling layer as a lambda function:
```
	def sampling(args):
	  mu, sigma = args
	  batch     = K.shape(mu)[0]
	  dim       = K.int_shape(mu)[1]
	  eps       = K.random_normal(shape=(batch, dim))
	  return mu + K.exp(sigma / 2) * eps


	z = Lambda(sampling, output_shape=(latent_dim,))([mu, sigma])
```
but then changed it to be able to run it with the corresponding keras version. The way it is defined now is the correct way. Also, note that the latest up-to-date implementation of the model will be available in the [rapidae](https://github.com/NahuelCostaCortez/rapidae) library, which is still under development, but I believe it will be ready at the end of this month.

Regards,
Nahuel

**Comment by nightwork23** [comment]
Thank you for your prompt reply.

**Comment by nightwork23** [comment]
Hello, sorry for the trouble again. After a few days of trying, I still have two questions now.

1.https://github.com/NahuelCostaCortez/Remaining-Useful-Life-Estimation-Variational/blob/main/RULRVE.ipynb
For the baseline, I have configured EarlyStopping and ModelCheckpoint callbacks. I run it until 10000 epochs and save the model for testing. The results show that the model will be overfitted at approximately 100 epochs. The best result is:test set RMSE:13.37,SCORE:304.91, which seems no obvious improvement.
I also found that the picture of the latent space was always the opposite compared to the results in the paper. Even training the model to overfit does not improve the situation.
![latent_space_epoch28test](https://github.com/NahuelCostaCortez/Remaining-Useful-Life-Estimation-Variational/assets/50656181/7cf90a10-7c89-42df-9de8-1a315ab41dcf)

2.As for the results for FD002, I have tried with ['s_3', 's_4', 's_7', 's_10', 's_11', 's_12'] and use the same hyper-parameters for the  FD003, the result is still bad:
test set RMSE:25.99,SCORE:15440,

I would appreciate it if you could give me some advice to help reproduce your work.

**Comment by NahuelCostaCortez** [comment]
As for your first question I´m not sure if I´m getting it right. You are reporting results better than the ones reported in the paper, so that´s a good sign.

The different distribution in the latent space is normal to happen. The network tries to find the latent space that allows to make the best predictions, so depending on how it starts the optimization it can place the healthiest samples in the leftmost part or in the rightmost part. If several training runs are repeated the latent space will change unless you fix all seeds.

As for the second question, yeah, that is indeed weird. 25.99 seems like a lot. I will look into it.

Also, keep in mind that although you should get as close as possible, you will not get the same numbers as in the paper because the VAE is a probabilistic model, so in each run the sampling of the normal will be different. This is fixed in keras 3 with the seed_generator function, that's why I am integrating this and other models in [rapidae](https://github.com/NahuelCostaCortez/rapidae).

**Comment by nightwork23** [comment]
Thank you for your reminding. I have run the code many times and found that the direction drawn by the latent space has not changed. Perhaps I should pay attention to the seeds problem.

As for the problem for FD002, your suggestions would be greatly appreciated.

**Comment by NahuelCostaCortez** [comment]
I just updated the rapidae package. You can check [this](https://github.com/NahuelCostaCortez/rapidae/blob/main/examples/rve_RUL.ipynb). I just executed it changing FD004 by FD002 and these are the results I get (better than those reported in the paper actually!):

![image](https://github.com/NahuelCostaCortez/Remaining-Useful-Life-Estimation-Variational/assets/43174126/3694c5cc-3308-41b4-84c1-9e5a03e5c545)

Hope this helps.

**Comment by nightwork23** [comment]
Thank you for your opinion.
